android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressButtons.kt

```diff
@@ -0,0 +1,128 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.gu.source.Source
+import com.gu.source.components.buttons.ButtonColours
+import com.gu.source.components.buttons.SourceBaseIconButton
+import com.gu.source.components.buttons.SourceButton
+import com.gu.source.daynight.AppColour
+import com.gu.source.daynight.AppColourMode
+import com.gu.source.icons.ChevronLeft
+import com.gu.source.icons.ChevronRight
+import com.gu.source.presets.palette.Neutral10
+import com.gu.source.presets.palette.Neutral100
+import com.gu.source.presets.palette.Neutral7
+import com.gu.source.utils.PreviewPhoneBothMode
+import com.gu.source.utils.PreviewTabletBothMode
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@SuppressLint("DiscouragedApi")
+@Composable
+fun ProgressButtons(
+    pagerState: PagerState,
+    buttonColours: ButtonColours = DefaultButtonColours,
+    prevButtonContentDescription: String?,
+    nextButtonContentDescription: String?,
+    modifier: Modifier = Modifier,
+    disabledButtonColours: ButtonColours? = disabledModeButtonColours(buttonColours),
+    pageSlideAnimationSpec: AnimationSpec<Float> = DefaultPageSlideAnimationSpec,
+) {
+    val coroutineScope = rememberCoroutineScope()
+
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
+    ) {
+        SourceBaseIconButton(
+            size = SourceButton.Size.Small,
+            buttonColours = buttonColours,
+            disabledButtonColours = disabledButtonColours,
+            onClick = {
+                handlePageTransition(
+                    coroutineScope,
+                    pagerState,
+                    ProgressDirection.Previous,
+                    pageSlideAnimationSpec
+                )
+            },
+            enabled = pagerState.canScrollBackward,
+            modifier = Modifier.offset(x = ProgressButtonTouchAdjustment * 2),
+        ) {
+            Icon(
+                imageVector = Source.Icons.Base.ChevronLeft,
+                contentDescription = prevButtonContentDescription,
+                modifier = it,
+            )
+        }
+        SourceBaseIconButton(
+            size = SourceButton.Size.Small,
+            buttonColours = buttonColours,
+            disabledButtonColours = disabledButtonColours,
+            onClick = {
+                handlePageTransition(
+                    coroutineScope,
+                    pagerState,
+                    ProgressDirection.Next,
+                    pageSlideAnimationSpec
+                )
+            },
+            enabled = pagerState.canScrollForward,
+        ) {
+            Icon(
+                imageVector = Source.Icons.Base.ChevronRight,
+                contentDescription = nextButtonContentDescription,
+                modifier = it,
+            )
+        }
+    }
+}
+
+fun handlePageTransition(
+    coroutineScope: CoroutineScope,
```

Comment:
Passing around `CoroutingScope` is a bad smell. Instead make this a `suspend fun` and use the `coroutineScope.launch` where you're calling this.

Nit: Also make this `private`

We could also rename this to `animateScrollToNext` because that's what it's doing - it determines the next page based on `direction` and then animate scrolls to it.